src/components/switch.js

```diff
@@ -0,0 +1,120 @@
+import React from 'react';
+import styled from '@emotion/styled';
+import { css, Input } from 'theme-ui';
+import { applyVariation as getFromThemeObj, getThemeStyles } from './../utils/getStyles';
+
+const defaultVariants = {
+  sm: {
+    switchBody: {
+      width: '30px',
+      height: '15px',
+      borderColor: 'borderGray',
+      bg: 'borderGray',
+      '&:checked': {
+        bg: 'error',
+      },
+    },
+    switchHandle: {
+      borderColor: 'borderGray',
+      bg: 'white',
+      '&:checked': {
+        bg: 'primary',
+      },
+    },
+  },
+  md: {
+    switchBody: {
+      width: '36px',
+      height: '18px',
+      borderColor: 'success',
+      bg: 'success',
+      '&:checked': {
+        borderColor: 'primary',
+        bg: 'primary',
+      },
+    },
+    switchHandle: {
+      borderColor: 'borderGray',
+      bg: 'white',
+      '&:checked': {
+        bg: 'primary',
+      },
+    },
+  },
+  lg: {
+    switchBody: {
+      width: '44px',
+      height: '22px',
+      borderColor: 'warning',
+      bg: 'warning',
+      '&:checked': {
+        borderColor: 'successDark',
+        bg: 'success',
+      },
+    },
+    switchHandle: {
+      borderColor: 'borderGray',
+      bg: 'white',
+      '&:checked': {
+        bg: 'warningDark',
+      },
+    },
+  },
+};
+
+const getWidth = (theme, variant, key = 'width') =>
```

Comment:
- [ ] you are accepting `key` parameter and retrieving value based on it. Then why the function is named `getWidth`?

Reply:
@a-saran